public/search/sql_async_search_interceptor.ts

```diff
@@ -77,18 +87,29 @@ export class SQLAsyncSearchInterceptor extends SearchInterceptor {
     };
 
     const onPollingSuccess = (pollingResult: any) => {
-      if (pollingResult && pollingResult.body.meta.status === 'SUCCESS') {
-        return false;
-      }
-      if (pollingResult && pollingResult.body.meta.status === 'FAILED') {
-        const jsError = new Error(pollingResult.data.error.response);
-        this.deps.toasts.addError(jsError, {
-          title: i18n.translate('queryEnhancements.sqlQueryError', {
-            defaultMessage: 'Could not complete the SQL async query',
-          }),
-          toastMessage: pollingResult.data.error.response,
-        });
-        return false;
+      if (pollingResult) {
+        const statusStr: string = (pollingResult.body.meta.status as string).toUpperCase();
+        const status = SparkJobState[statusStr as keyof typeof SparkJobState];
+        switch (status) {
+          case SparkJobState.SUCCESS:
+            return false;
```

Comment:
unrelated to this PR but thought I get your feedback on this function.

when i refactored this a little I kept intact like the returning of a boolean but it returns false because the rxjs operator `takeWhile` needs to get the false value to stop polling. 

but it's confusing to read do you think it's worth to rename this function? or shift the logic so that in the utils function it negates the value here? Because return false when `SUCCESS` from onPollingSuccess could get confusing.

Reply:
It could be confusing at first, but I think it's generally derivable that the usage is to return `false` to halt polling and `true` to continue.

If we want to make it more clear I think the way to go would be to refactor the polling util to rely on a polling state enum, and return values like `Polling.DONE`, `Polling.CONTINUE`, `Polling.ERROR`. Could even make it more sophisticated with the specific statuses, having things like `TIMEOUT` or `ABORT`.

Reply:
Not sure if it's obvious by me introducing `JobState` to begin with, but I'm a big fan of using state machines and strong typing (i.e. lots of enums) to keep track of program state and handle errors.